ls1intum/artemis-android
# `Feature`: Save messages for later

### Problem Description
The Android app did not yet support the "Saved Messages" feature recently introduced in the webapp. This PR introduces the related use cases. 

### Changes
 
- Created new network service `SavedPostService`
  - Extended `BasePost` with new  `SavedPost` 
- Created actions to mark/unmark a post as saved
-  Created new section in the ConversationOverview for Saved messages
- Introduced new `SavedPostsScreen` for displaying the list of saved posts
  - Implemented design heavily inspired by the webapp
  - Pressing on a saved post navigates the user to the post's thread
  - Long pressing on a post shows a BottomSheet with action to change the status
  - Introduced `MetisModificationTaskHandler` for sharing MetisModification functionality 
  - Changed core of `PostItem` to be re-used by `SavedPostItem`
- Created tests

### Steps for testing
- Go to any chat
  - Long press on a post to show to action bottomSheet and save it for later -> Post is highlighted in blue and a "Saved for later" indicator is shown
  - Click on post to show it in the thread view
  - Here you can also change the `isSaved` via the ActionTabBar
- Go back to the conversationOverview 
  - Notice the new "Saved messages" section
  - Click on "In progress" -> saved messages are shown
  - Click on the saved post -> navigate to the post's thread
  - Navigate back
  - Change the savedPostStatus of the post via the action bottomSheet

### Screenshots

https://github.com/user-attachments/assets/cf8bf216-cdb1-402b-8873-feada6773088



### Known Issues
- Navigating back after clicking on a saved message, currently does not take the user back to the SavedPostScreen -> tackled in https://github.com/ls1intum/artemis-android/pull/280
- When toggling the `isSaved` state in the ThreadUi via the ActionTabBar, the action icons flicker -> #268 
- SavedMessages does not yet support the split screen layout with the conversation overview on the left (eg on tablet) -> #288
- Spamming the savePost button in the Threads ActionTabBar results in multiple saved posts being created for that post -> #307 

**Further Input needed** :  Sometimes I could observe that when navigating to a post via the SavedPostScreen that the thread failed to load. Even after multiple retries the thread only very shortly showed up but then disappeared again. I could not consistently reproduce this issue, so if you find a scenario where this occurs regularly, please let me know
